lib/presentation/my_listings/my_listings_view.dart

```diff
@@ -0,0 +1,313 @@
+                    onRefresh: () async {
+                      await context.read<MyListingsCubit>().refreshListings();
+                    },
+                    child: BlocBuilder<MyListingsCubit, MyListingsState>(
+                      builder: _buildBody,
+                    ),
+                  ),
+                  // Top scroll shadow
+                  if (_isScrolled)
+                    Positioned(
+                      top: 0,
+                      left: 0,
+                      right: 0,
+                      child: Container(
+                        height: 20,
+                        decoration: BoxDecoration(
+                          gradient: LinearGradient(
+                            begin: Alignment.topCenter,
+                            end: Alignment.bottomCenter,
+                            colors: [
+                              AppColors.black.withValues(alpha: 0.8),
+                              AppColors.black.withValues(alpha: 0.4),
+                              AppColors.black.withValues(alpha: 0.0),
+                            ],
+                          ),
+                        ),
+                      ),
+                    ),
+                ],
+              ),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  @override
+  void dispose() {
+    _scrollController.dispose();
+    _searchController.dispose();
+    super.dispose();
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    _scrollController.addListener(_onScroll);
+  }
+
+  Widget _buildBody(BuildContext context, MyListingsState state) {
+    if (state.isLoading) {
+      return Padding(
+        padding: const EdgeInsets.all(16),
+        child: SingleChildScrollView(
+          child: Column(
+            children: List.generate(
+              5, // Show 5 shimmer cards
+              (index) => const ShimmerListingItemView(),
+            ),
+          ),
+        ),
+      );
+    }
+
+    if (state.listings.isEmpty && !state.isLoading) {
+      return Center(
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: [
+            const Icon(Icons.inbox_outlined, size: 64, color: AppColors.grey),
+            const SizedBox(height: 16),
+            Text(
+              state.status == ApiConstants.queryMarketStatusLive
+                  ? I18n.of(context).myListingsNoActiveListings
+                  : I18n.of(context).myListingsNoReservedListings,
+              style: const TextStyle(
+                color: AppColors.grey,
+                fontSize: 18,
+                fontWeight: FontWeight.w500,
+              ),
+            ),
+            const SizedBox(height: 8),
+            Text(
+              state.status == ApiConstants.queryMarketStatusLive
+                  ? I18n.of(context).myListingsNoActiveListingsDescription
+                  : I18n.of(context).myListingsNoReservedListingsDescription,
```

Comment:
Add logic if there is a search query then it should show no results found instead of no active listings/reserved items